src/libp2p/Libp2p/Libp2pPeerFactoryBuilder.cs

```diff
@@ -21,13 +25,17 @@ public ILibp2pPeerFactoryBuilder WithPlaintextEnforced()
 
     public Libp2pPeerFactoryBuilder(IServiceProvider? serviceProvider = default) : base(serviceProvider)
     {
+        _loggerFactory = serviceProvider?.GetService(typeof(ILoggerFactory)) as ILoggerFactory;
+        _multiplexerSettings = serviceProvider?.GetService(typeof(MultiplexerSettings)) as MultiplexerSettings;
+
     }
 
     protected override ProtocolStack BuildStack()
     {
+
         ProtocolStack tcpEncryptionStack = enforcePlaintext ?
             Over<PlainTextProtocol>() :
-            Over<NoiseProtocol>();
+         Over<NoiseProtocol>().Or(new TlsProtocol(_multiplexerSettings, _loggerFactory));
```

Comment:
Could you instead just pass class in generic parameter? Like for noise protocol.(but instead of over use Or. By doing it in such way we will not need factory _loggerFactory and _multiplexerSettings to be passed here.

Reply:
done